system-linux.c

```diff
@@ -1753,6 +1753,37 @@ system_set_ethtool_settings(struct device *dev, struct device_settings *s)
 	ioctl(sock_ioctl, SIOCETHTOOL, &ifr);
 }
 
+static void
+system_set_ethtool_eee_settings(struct device *dev, struct device_settings *s)
+{
+	struct ethtool_eee eeecmd = {
+		.cmd = ETHTOOL_GEEE,
+	};
+	struct ifreq ifr = {
+		.ifr_data = (caddr_t)&eeecmd,
+	};
+
+	strncpy(ifr.ifr_name, dev->ifname, sizeof(ifr.ifr_name) - 1);
+
+	if (ioctl(sock_ioctl, SIOCETHTOOL, &ifr) != 0)
```

Comment:
Why is this redundant `ETHTOOL_GEEE` operation required? Was this just cargo culted from `system_set_ethtool_settings()` ?

To me it seems as if the function implementation can be reduced to just this:
```c
static void
system_set_ethtool_eee_settings(struct device *dev, struct device_settings *s)
{
	struct ethtool_eee eeecmd = {
		.cmd = ETHTOOL_SEEE,
		.eee_enabled = s->eee,
	};

	struct ifreq ifr = {
		.ifr_data = (caddr_t)&eeecmd,
	};

	strncpy(ifr.ifr_name, dev->ifname, sizeof(ifr.ifr_name) - 1);

	if (ioctl(sock_ioctl, SIOCETHTOOL, &ifr) != 0)
		netifd_log_message(L_WARNING, "cannot set eee %d for device %s", bit, dev->ifname);
}
```

Reply:
more of a cargo from ethtool do_ssee, thanks for pointing this out, updated as suggested.